compiler/rustc_resolve/src/lib.rs

```diff
@@ -1022,6 +1023,7 @@ pub struct Resolver<'ra, 'tcx> {
 
     /// N.B., this is used only for better diagnostics, not name resolution itself.
     field_names: LocalDefIdMap<Vec<Ident>>,
+    field_defaults: LocalDefIdMap<Vec<Symbol>>,
```

Comment:
Using `Symbol` instead `Ident` raises a few warning flags in my head because I'm pretty sure that this means anything involving `field_defaults` doesn't handle macro hygiene correctly. This is only relevant to decl macros 2.0 (feature `decl_macro`).

It might be a non-issue, I haven't thought that hard yet. I'll need to read the rest of this PR first.